cargo-paradedb/README.md

```diff
@@ -0,0 +1,86 @@
+# ParadeDB Cargo Dev Tool
+
+## Installation
+
+The first time you install `cargo-paradedb`, you should navigate to the `cargo-paradedb` crate and run:
+
+```sh
+cargo run install
+```
+
+After this first-time installation, you can run `cargo paradedb install` from anywhere and Cargo will globally re-install `cargo-paradedb` with the latest code changes from your source folder.
+
+If you don't want to install globally, you can always just `cargo run` from the `cargo-paradedb` crate folder.
+
+### Installing From Git Url
+
+In containers or cloud instances, it's useful to be able to install globally with a single command:
+
+```sh
+cargo install --git https://github.com/paradedb/paradedb.git cargo-paradedb
+```
+
+This will install the tool for use as `cargo paradedb` without having to clone the repository first. You can also specify a branch:
+
+```sh
+cargo install \
+    --git https://github.com/paradedb/paradedb.git \
+    --branch new-feature-branch \
+    cargo-paradedb
+```
+
+## Benchmarks
+
+To run benchmarks, you must be able to connect to a running ParadeDB instance. `cargo-paradedb` accepts a Postgres connection url in one of the following ways:
```

Comment:
The way to run them is therefore to either spin up a ParadeDB Docker container or have a pgrx Postgres running?

Reply:
Yes, right now this tool doesn't try and manage the instances of anything that we benchmark, interacting with Postgres/ES through URLs.

I think this is the best way to constrain its scope for now. There's just too many different variables to setting up Postgres in different environments. We don't want the benchmarks to be locked to docker containers for the same reasons we don't want that for our tests. You also need different run commands based on where Postgres installed, whether its on a Docker volume, etc.

That still wouldn't make getting everything up and running a one-click thing, as deployments on EC2 etc. require volume mount, permission changes, network config, etc. It just doesn't seem like something I should try to automate right now.

That being said... this tool is here to be extended in all kinds of ways. It's not just for benchmarks. If we want to come up with shortcut commands for long cumbersome `docker run ...` incantations and the like, we should liberally add them to `cargo-paradedb`.

I just don't think we should bake any of that into how we run the benchmarks. Running tests/benchmarks should be maximally flexible.